contracts/deployer/AccessControlledDeployer.sol

```diff
@@ -0,0 +1,139 @@
+// Copyright Immutable Pty Ltd 2018 - 2024
+// SPDX-License-Identifier: Apache 2.0
+pragma solidity 0.8.19;
+
+import {IDeployer} from "@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IDeployer.sol";
+import {Pausable} from "@openzeppelin/contracts/security/Pausable.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {AccessControlEnumerable} from "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
+
+contract AccessControlledDeployer is AccessControlEnumerable, Pausable {
+    /// @notice Role identifier for those who can pause the deployer
+    bytes32 public constant PAUSER_ROLE = keccak256("PAUSER");
+
+    /// @notice Role identifier for those who can unpause the deployer
+    bytes32 public constant UNPAUSER_ROLE = keccak256("UNPAUSER");
+
+    /// @notice Role identifier for those who can deploy contracts
+    bytes32 public constant DEPLOYER_ROLE = keccak256("DEPLOYER");
+
+    /**
+     * @notice Construct a new RBACDeployer contract
+     * @param admin The address to grant the DEFAULT_ADMIN_ROLE
+     * @param pauser The address to grant the PAUSER_ROLE
+     * @param unpauser The address to grant the UNPAUSER_ROLE
+     */
+    constructor(address admin, address pauser, address unpauser) {
+        require(admin != address(0), "admin is the zero address");
+        require(pauser != address(0), "pauser is the zero address");
+        require(unpauser != address(0), "unpauser is the zero address");
+
+        _setupRole(DEFAULT_ADMIN_ROLE, admin);
+        _setupRole(PAUSER_ROLE, pauser);
+        _setupRole(UNPAUSER_ROLE, unpauser);
+    }
+
+    /**
+     * @notice Deploys a contract using a deployment method defined by `deployer`
+     * @param deployer The create2 or create3 deployer contract that will deploy the contract
+     * @param bytecode The bytecode of the contract to be deployed
+     * @param salt A salt to influence the contract address
+     * @dev Only address with DEPLOYER_ROLE can call this function
+     * @dev The function can only be called if the contract is not in a paused state
+     * @return The address of the deployed contract
+     */
+    function deploy(IDeployer deployer, bytes memory bytecode, bytes32 salt)
+        external
+        payable
+        whenNotPaused
+        onlyRole(DEPLOYER_ROLE)
+        returns (address)
+    {
+        require(address(deployer) != address(0), "deployer contract is the zero address");
+        return deployer.deploy(bytecode, salt);
```

Comment:
Is tx.value implicitly passed to subsequent calls? I thought this would need to be 
```suggestion
        return deployer.deploy{value: msg.value}(bytecode, salt);
```